src/light_client.rs

```diff
@@ -50,11 +53,75 @@ use crate::{
 	types::{self, BlockVerified, LightClientConfig},
 };
 
+#[async_trait]
+#[automock]
+pub trait LightClient {
+	async fn fetch_cells_from_dht(
+		&self,
+		positions: &[Position],
+		block_number: u32,
+	) -> Result<(Vec<Cell>, Vec<Position>)>;
+	async fn insert_cells_into_dht(&self, block: u32, cells: Vec<Cell>) -> Result<f32>;
+	async fn insert_rows_into_dht(&self, block: u32, rows: Vec<(RowIndex, Vec<u8>)>)
+		-> Result<f32>;
+	async fn get_kate_proof(&self, hash: H256, positions: &[Position]) -> Result<Vec<Cell>>;
+	fn store_block_header_in_db(&self, header: &Header, block_number: u32) -> Result<()>;
+	fn store_confidence_in_db(&self, count: u32, block_number: u32) -> Result<bool>;
+}
+
+#[derive(Clone)]
+pub struct LightClientImpl {
+	db: Arc<DB>,
+	network_client: Client,
+	rpc_client: OnlineClient<AvailConfig>,
+}
+
+#[async_trait]
+impl LightClient for LightClientImpl {
+	async fn insert_cells_into_dht(&self, block: u32, cells: Vec<Cell>) -> Result<f32> {
+		Ok(self
+			.network_client
+			.insert_cells_into_dht(block, cells)
+			.await)
```

Comment:
Do we need to wrap `f32` with `Result` since there is no error case? We should probably change signature to 
`async fn insert_cells_into_dht(&self, block: u32, cells: Vec<Cell>) -> f32`.